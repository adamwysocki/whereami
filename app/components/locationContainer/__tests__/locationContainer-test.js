// __tests__/locationContainer-test.js
import 'react-native';
import React from 'react';
import LocationContainer from '../index';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

var onPress = function() {
  return;
}

describe('locationContainer', () => {
  const fetch = global.fetch

  beforeEach(() => {
    global.fetch = jest.fn().mockImplementation(() =>
        Promise.resolve(new Response('{"results": [ {"address_components": [ {}, {}, {"long_name": "Linden"}, {}, {"long_name": "New Jersey"}, {"short_name": "US"} ] }]}')));
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <LocationContainer
        onPress={onPress}
        latitude={31}
        longitude={-58}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
      global.fetch = fetch;
  });
});
