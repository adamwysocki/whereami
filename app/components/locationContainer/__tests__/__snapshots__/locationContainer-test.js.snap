exports[`locationContainer renders correctly 1`] = `
<div
  className="View"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
      "marginTop": 70
    }
  }>
  <div
    className="Text"
    style={
      Object {
        "color": "#000000",
        "fontSize": 22,
        "fontWeight": "900",
        "marginBottom": 70,
        "textAlign": "center"
      }
    }>
    unknown
    , 
    unknown
     
    unknown
  </div>
  <div
    className="TouchableHighlight"
    onClick={[Function onPress]}
    onPress={[Function onPress]}>
    <div
      className="View"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0000FF",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "width": 250
        }
      }>
      <div
        className="Text"
        style={
          Object {
            "color": "white",
            "fontWeight": "900",
            "textAlign": "center"
          }
        }>
        RESET
      </div>
    </div>
  </div>
</div>
`;
